### ISSUE TITLE
`InfluxQLQueryAPI` isn't compatible with InfluxDB 1.8+

### ISSUE THREAD

**Issue by btasker** [influxdata/influxdb-client-java#458]
In https://github.com/influxdata/influxdb-client-java/issues/365 / https://github.com/influxdata/influxdb-client-java/issues/366 new functionality was added to allow InfluxQL queries to be run against the compatibility API using the client.

However, this change is not compatible with Influx 1.8.x because of an implementation detail in `InfluxQLQueryService`

That interface does the following
```
	@Headers({ "Accept:application/csv", "Content-Type:application/vnd.influxql" })
	@POST("query")
	Call<ResponseBody> query(
			@Body String query,
			@Nonnull @Query("db") String db,
			@Query("rp") String retentionPolicy,
			@Query("epoch") String epoch,
			@Header("Zap-Trace-Span") String zapTraceSpan
	);
```
This sends the InfluxQL query as the request body, with a content-type of `application/vnd.influxql`

This is broadly equivalent to doing
```
curl -H "Authorization: Token $MYTOK" -d "select last(download) from internet_speed limit 1" -v "http://192.168.3.5:18086/query?db=telegraf" -H "Content-Type: application/vnd.influxql"
```

However, InfluxDB 1.8.x does not support this, and requires that the query be sent with the query in `q`
```
ben@bumblebee:~$ curl -H "Authorization: Token $MYTOK" -d "select last(download) from internet_speed limit 1" -v "http://192.168.3.84:8086/query?db=telegraf" -H "Content-Type: application/vnd.influxql"
*   Trying 192.168.3.84:8086...
* Connected to 192.168.3.84 (192.168.3.84) port 8086 (#0)
> POST /query?db=telegraf HTTP/1.1
> Host: 192.168.3.84:8086
> User-Agent: curl/7.81.0
> Accept: */*
> Authorization: Token <redacted>
> Content-Type: application/vnd.influxql
> Content-Length: 49
> 
* Mark bundle as not supporting multiuse
< HTTP/1.1 400 Bad Request
< Content-Type: application/json
< Request-Id: 3775f635-5456-11ed-8071-0242ac130002
< X-Influxdb-Build: OSS
< X-Influxdb-Error: missing required parameter "q"
< X-Influxdb-Version: 1.8.10
< X-Request-Id: 3775f635-5456-11ed-8071-0242ac130002
< Date: Tue, 25 Oct 2022 11:14:39 GMT
< Content-Length: 45
< 
{"error":"missing required parameter \"q\""}
```
2.x can support either format, so it would probably make most sense to adjust the interface to send a form-encoded request body.

__Steps to reproduce:__
List the minimal actions needed to reproduce the behavior.

1. Run InfluxDB 1.8.10
2. Use `getInfluxQLQueryApi()` to get the API
3. Run an InfluxQL query
4. Receive `{"error":"missing required parameter \"q\""}`

__Expected behavior:__
The query should run

__Actual behavior:__
The query is not run

__Specifications:__
 - Client Version: 6.6.0
 - InfluxDB Version: 1.8.10
 - JDK Version: Any
 - Platform: Any

**Comment by bednar** [comment]
Hi @btasker,

thanks for using our client. 

Is this something you might be willing to help with?

Regards

**Comment by btasker** [comment]
Hi @bednar 

Yep, with luck (and time allowing) should have a PR up for you a little later today

**Comment by btasker** [comment]
The PR above switches to sending a form-encoded POST body.

Wireshark shows the following

Before change:
```
POST /query?db=testing_db&epoch=s HTTP/1.1
Accept: application/csv
User-Agent: influxdb-client-java/unknown
Authorization: Token my-token
Accept-Encoding: identity
Content-Type: application/vnd.influxql
Content-Length: 36
Host: 192.168.3.84:8086
Connection: Keep-Alive

SELECT FIRST("free") FROM "influxql"HTTP/1.1 400 Bad Request
Content-Type: application/csv
Request-Id: 05f910b7-5468-11ed-956e-0242ac130002
X-Influxdb-Build: OSS
X-Influxdb-Error: missing required parameter "q"
X-Influxdb-Version: 1.8.10
X-Request-Id: 05f910b7-5468-11ed-956e-0242ac130002
Date: Tue, 25 Oct 2022 13:22:07 GMT
Content-Length: 41

error
"missing required parameter ""q"""
```

After change
```
POST /query?db=testing_db&epoch=s HTTP/1.1
Accept: application/csv
User-Agent: influxdb-client-java/unknown
Authorization: Token my-token
Accept-Encoding: identity
Content-Type: application/x-www-form-urlencoded
Content-Length: 56
Host: 192.168.3.84:8086
Connection: Keep-Alive

q=SELECT%20FIRST%28%22free%22%29%20FROM%20%22influxql%22
```

Switching out the query to one appropriate to my schema yields data
```
POST /query?db=telegraf&epoch=s HTTP/1.1
Accept: application/csv
User-Agent: influxdb-client-java/unknown
Authorization: Token my-token
Accept-Encoding: identity
Content-Type: application/x-www-form-urlencoded
Content-Length: 65
Host: 192.168.3.84:8086
Connection: Keep-Alive

q=SELECT%20last%28%22download%22%29%20FROM%20%22internet_speed%22HTTP/1.1 200 OK
Content-Type: application/csv
Request-Id: ee40683f-5469-11ed-9e7e-0242ac130002
X-Influxdb-Build: OSS
X-Influxdb-Version: 1.8.10
X-Request-Id: ee40683f-5469-11ed-9e7e-0242ac130002
Date: Tue, 25 Oct 2022 13:35:46 GMT
Transfer-Encoding: chunked

41
name,tags,time,last
internet_speed,,1666703706,48.87309862652547

0
```

**Comment by btasker** [comment]
I've also run the same test against 2.x to ensure that the change doesn't break support for that

```
POST /query?db=telegraf&epoch=s HTTP/1.1
Accept: application/csv
User-Agent: influxdb-client-java/unknown
Authorization: Token <redacted>
Accept-Encoding: identity
Content-Type: application/x-www-form-urlencoded
Content-Length: 65
Host: 192.168.3.5:18086
Connection: Keep-Alive

q=SELECT%20last%28%22download%22%29%20FROM%20%22internet_speed%22HTTP/1.1 200 OK
Content-Type: application/csv
X-Influxdb-Build: OSS
X-Influxdb-Version: v2.3.0+SNAPSHOT.090f681737
Date: Tue, 25 Oct 2022 13:38:03 GMT
Content-Length: 65

name,tags,time,last
internet_speed,,1666704600,48.87309862652547
```

**Comment by bednar** [comment]
Closed by #459
